homeassistant/components/anova_sous_vide/sensor.py

```diff
@@ -0,0 +1,168 @@
+"""Support for Anova Sous Vide Sensors."""
+from __future__ import annotations
+
+from datetime import timedelta
+import logging
+
+from anova_wifi import AnovaOffline, AnovaPrecisionCookerSensor
+import async_timeout
+
+from homeassistant import config_entries
+from homeassistant.components.sensor import (
+    SensorDeviceClass,
+    SensorEntity,
+    SensorEntityDescription,
+    SensorStateClass,
+)
+from homeassistant.const import UnitOfTemperature, UnitOfTime
+from homeassistant.core import HomeAssistant, callback
+from homeassistant.helpers.entity import DeviceInfo, EntityCategory
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+from homeassistant.helpers.typing import StateType
+from homeassistant.helpers.update_coordinator import (
+    CoordinatorEntity,
+    DataUpdateCoordinator,
+    UpdateFailed,
+)
+
+from .const import DOMAIN
+
+_LOGGER = logging.getLogger(__name__)
+
+SENSOR_DESCRIPTIONS: dict[str, SensorEntityDescription] = {
+    AnovaPrecisionCookerSensor.COOK_TIME: SensorEntityDescription(
+        key=AnovaPrecisionCookerSensor.COOK_TIME,
+        state_class=SensorStateClass.TOTAL_INCREASING,
+        native_unit_of_measurement=UnitOfTime.SECONDS,
+        icon="mdi:clock-outline",
+        name="Cook Time",
+    ),
+    AnovaPrecisionCookerSensor.STATE: SensorEntityDescription(
+        key=AnovaPrecisionCookerSensor.STATE, name="State"
+    ),
+    AnovaPrecisionCookerSensor.MODE: SensorEntityDescription(
+        key=AnovaPrecisionCookerSensor.MODE, name="Mode"
+    ),
+    AnovaPrecisionCookerSensor.TARGET_TEMPERATURE: SensorEntityDescription(
+        key=AnovaPrecisionCookerSensor.TARGET_TEMPERATURE,
+        native_unit_of_measurement=UnitOfTemperature.CELSIUS,
+        device_class=SensorDeviceClass.TEMPERATURE,
+        state_class=SensorStateClass.MEASUREMENT,
+        icon="mdi:thermometer",
+        name="Target Temperature",
+    ),
+    AnovaPrecisionCookerSensor.COOK_TIME_REMAINING: SensorEntityDescription(
+        key=AnovaPrecisionCookerSensor.COOK_TIME_REMAINING,
+        native_unit_of_measurement=UnitOfTime.SECONDS,
+        icon="mdi:clock-outline",
+        name="Cook Time Remaining",
+    ),
+    AnovaPrecisionCookerSensor.FIRMWARE_VERSION: SensorEntityDescription(
+        key=AnovaPrecisionCookerSensor.FIRMWARE_VERSION,
+        entity_category=EntityCategory.DIAGNOSTIC,
+        name="Firmware Version",
+    ),
```

Comment:
This is the firmware version of the whole device as far as I understand. Maybe it's better to use it in the device info (`sw_version`) instead of a separate sensor entity? Also, probably it does not change regularly, so there is no need to poll it each 30 seconds as other sensors